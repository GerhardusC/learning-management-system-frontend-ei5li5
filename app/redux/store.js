"use client";
import { configureStore } from "@reduxjs/toolkit";
import creatorEditingSlice from "./slices/creatorEditingSlice";
import learnerLessonProgressSlice from "./slices/learnerLessonProgressSlice";
import alertStateSlice from "./slices/alertStateSlice";
// Generic store containing all slices.
export const store = configureStore({
  reducer: {
    creatorEditingState: creatorEditingSlice,
    learnerLessonProgressState: learnerLessonProgressSlice,
    alertState: alertStateSlice,
  },
});
