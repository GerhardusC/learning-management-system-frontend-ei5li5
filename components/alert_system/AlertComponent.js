"use client";
import React from "react";
import { useSelector } from "react-redux";
import TempAlert from "./TempAlert";
import PromptAlert from "./PromptAlert";

// This component returns a different alert depending on the selected
// alert type. Currently there are temporary alerts and prompting alerts.
// I.e. alerts that disappear after some time and alerts that need to be
// dismissed.
// Currently in the program only temp alerts are being used, but the prompt
// alerts do work.
const AlertComponent = () => {
  const alertStatus = useSelector((state) => state.alertState.alertStatus);
  const alertType = useSelector((state) => state.alertState.alertType);
  const alertMessage = useSelector((state) => state.alertState.alertMessage);
  if (alertStatus) {
    switch (alertType) {
      case "temp":
        return <TempAlert message={alertMessage} />;
      case "prompt":
        return <PromptAlert message={alertMessage} />;
      default:
        return;
    }
  } else {
    return;
  }
};

export default AlertComponent;
