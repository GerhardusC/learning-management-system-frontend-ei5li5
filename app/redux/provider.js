"use client";
import { Provider } from "react-redux";
import { store } from "./store";
// The providers have to be built like this in the Next.js app router.
export const Providers = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
