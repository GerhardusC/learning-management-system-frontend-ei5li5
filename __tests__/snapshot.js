import { render } from "@testing-library/react";
import AlertComponent from "@/components/alert_system/AlertComponent";
import { Providers } from "@/app/redux/provider";
it("renders alert correctly", () => {
  const { container } = render(
    <Providers>
      <AlertComponent />
    </Providers>
  );
  expect(container).toMatchSnapshot();
});
