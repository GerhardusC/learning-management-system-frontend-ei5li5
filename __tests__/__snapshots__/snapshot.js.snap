// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders alert correctly 1`] = `<div />`;
